components/brave_sync/time_limited_words_unittest.cc

```diff
@@ -0,0 +1,185 @@
+      []() { return g_overridden_now; }, nullptr, nullptr);
+}
+}  // namespace
+
+TEST(TimeLimitedWordsTest, GetRoundedDaysDiff) {
+  const base::Time time1 = QrCodeDataValidator::GetQRv1SunsetDay();
+
+  base::Time time2 = time1 + base::Hours(11);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time1, time2), 0);
+
+  time2 = time1 + base::Hours(13);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time1, time2), 1);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time2, time1), -1);
+}
+
+TEST(TimeLimitedWordsTest, GetIndexByWord) {
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("abandon"), 0);
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("ability"), 1);
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("not_bip39_word"), -1);
+}
+
+TEST(TimeLimitedWordsTest, GetWordByIndex) {
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(0), "abandon");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(1), "ability");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(2047), "zoo");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(2048), "abandon");
+}
+
+TEST(TimeLimitedWordsTest, GenerateForDate) {
+  EXPECT_EQ(std::string(kValidSyncCode) + " abandon",
+            TimeLimitedWords::GenerateForDate(
+                kValidSyncCode, QrCodeDataValidator::GetQRv1SunsetDay()));
+  EXPECT_EQ(std::string(kValidSyncCode) + " ability",
+            TimeLimitedWords::GenerateForDate(
+                kValidSyncCode,
+                QrCodeDataValidator::GetQRv1SunsetDay() + base::Days(1)));
+  EXPECT_EQ("", TimeLimitedWords::GenerateForDate(
+                    kValidSyncCode,
+                    QrCodeDataValidator::GetQRv1SunsetDay() - base::Days(1)));
+}
+
+TEST(TimeLimitedWordsTest, Validate) {
+  std::string pure_words;
+  WordsValidationResult result;
+
+  {
+    // Valid v1 sync code, prior to sunset date
+    auto time_override = OverrideWithTimeNow(
+        TimeLimitedWords::GetWordsV1SunsetDay() - base::Days(1));
+    result = TimeLimitedWords::Validate(kValidSyncCode, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kValid);
+    EXPECT_EQ(pure_words, kValidSyncCode);
+  }
+
+  {
+    // Valid v1 sync code plus ending space, prior to sunset date
+    auto time_override = OverrideWithTimeNow(
+        TimeLimitedWords::GetWordsV1SunsetDay() - base::Days(1));
+    result = TimeLimitedWords::Validate(kValidSyncCode + std::string(" "),
+                                        &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kValid);
+    EXPECT_EQ(pure_words, kValidSyncCode);
+  }
+
+  {
+    // Invalid v1 sync code, prior to sunset date
+    auto time_override = OverrideWithTimeNow(
+        TimeLimitedWords::GetWordsV1SunsetDay() - base::Days(1));
+    result = TimeLimitedWords::Validate(kInvalidSyncCode, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kNotValidPureWords);
+    EXPECT_EQ(pure_words, "");
+  }
+
+  const base::Time anchorDayForWordsV2 =
+      QrCodeDataValidator::GetQRv1SunsetDay() + base::Days(20);
```

Comment:
Here I would not use the sunset date as the base date because V2 codes should be valid before the sunset date. It should be a date after the **epoch** instead.

Reply:
thanks, I fixed this at https://github.com/brave/brave-core/pull/12300/commits/4c59158b9070eacf74afa52c42c1930b85f942e4